libtenzir/src/execution_node.cpp

```diff
@@ -269,18 +264,32 @@ struct exec_node_state {
     TENZIR_DEBUG("{} {} shut down", *self, op->name());
     instance.reset();
     ctrl.reset();
-    metrics->emit();
+    metrics->emit(paused);
     if (demand and demand->rp.pending()) {
       demand->rp.deliver();
     }
   }
 
+  template <class... Duration>
+    requires(std::is_same_v<Duration, duration> && ...)
+  auto make_timer_guard(Duration&... elapsed) {
+    return caf::detail::make_scope_guard(
+      [&, start_time = std::chrono::steady_clock::now()] {
+        if (paused) {
+          // Do not add a delta to paused pipeline metrics.
+          return;
```

Comment:
This only affects the `pause` and `resume` message handlers: technically, they should still count towards scheduled time.

Reply:
So the `pause` and `resume` handlers should force a delta? Or should every `time_scheduled` delta be recorded, regardless of whether the node has been paused before, since `time_scheduled` also records the time for pushing and pulling?

Reply:
I think we can ignore this for now. It's not a big difference, not worth putting much time into this now.